lib/src/types/c2_full_results.dart

```diff
@@ -0,0 +1,43 @@
+// ignore_for_file: invalid_annotation_target
+
+part of 'index.dart';
+
+@freezed
+class C2FullResults with _$C2FullResults {
+  //TODO: figure out how to get this into JSON as time_formatted
+  // String get timeFormatted => Duration(
+  //         seconds: this.time.toInt(),
+  //         milliseconds:
+  //             (this.time.remainder(1) * Duration.millisecondsPerSecond).toInt())
+  //     .toString();
+
+  C2FullResults._();
+
+  factory C2FullResults({
+    @JsonKey(name: 'id') @Default(0) int id,
+    @JsonKey(name: 'user_id') @Default(0) int userId,
+    @JsonKey(name: 'date') @TimestampConverter() required DateTime date,
+    @JsonKey(name: 'timezone') String? timezone,
+    @JsonKey(name: 'date_utc') @TimestampOrNullConverter() DateTime? dateUtc,
+    @JsonKey(name: 'distance') @Default(0) int distance,
+    @JsonKey(name: 'type') @Default(C2ResultType.rower) C2ResultType type,
+    @JsonKey(name: 'time') @DecimalIntConverter.tenths() required double time,
+    @JsonKey(name: 'workout_type') @Default(C2APIWorkoutType.JustRow)
+    C2APIWorkoutType workoutType,
+    @JsonKey(name: 'source') @Default("c2logbook dart") String source,
+    @JsonKey(name: 'weight_class') @Default(C2WeightClass.heavyweight)
+    C2WeightClass weightClass,
+    @JsonKey(name: 'verified') @Default(false) bool verified,
+    @JsonKey(name: 'ranked') @Default(false) bool ranked,
+    @JsonKey(name: 'comments') String? comments,
+    @JsonKey(name: 'privacy') @Default(C2PrivacyLevel.private) C2PrivacyLevel privacy,
+    @JsonKey(name: 'rest_time') @DecimalIntConverter.tenths() double? restTime,
+    @JsonKey(name: 'stroke_rate') int? strokeRate,
+    @JsonKey(name: 'heart_rate') @Default(null) C2HeartRate? heartRate,
+    @JsonKey(name: 'workout') @Default(null) C2Workout? workout,
```

Comment:
was there an issue with the json keys for fields that dont have multiword names (like `workout`) where they werent getting recognized without making it explicit? or were these just added back for code cleanliness/formatting consistency?

I noticed when testing that if the key and the field name are the same, the `@JsonKey` isnt necessary to make explicit

Reply:
some fields werent making it in. adding them all fixed it